README.md

```diff
@@ -28,6 +28,46 @@ libraryDependencies ++= {
 `scala-2.12-junit-mixin-plugin` is platform-independent.
 It works with Scala on the JVM and with Scala.js.
 
+## How it works
+
+The compiler plugin performs a very simple job.
+When a JUnit-related method is inherited by a class from an interface as a
+default method, an explicit forwarder is added in the class to call the default
+method.
+
+For example, consider the following Scala code:
+
+```scala
+trait MyTestsBase {
+  @Test def add(): Unit =
+    assertEquals(3, 1 + 2)
+
+  def notATest(): Unit =
+    assertEquals(5, 2 + 3)
+}
+
+class MyTests extends MyTestsBase
+```
+
+Scala 2.12.0-M5 identifies that `add()` does not need a forward in `MyTests`,
+because it will be inherited as a default method on the JVM.
+But this fools JUnit 4, which does not look for `@Test` methods in interfaces.
+In this case, the plugin requires `MyTests` as follows:
```

Comment:
Er that was meant to be "rewrites" ... No idea how that became "requires" between my brain and my fingers ...